internal/martian/flush.go

```diff
@@ -23,7 +23,7 @@ import (
 )
 
 func shouldFlush(res *http.Response) bool {
-	if res.Request.Method == http.MethodHead {
+	if res.Request.Method == http.MethodHead || res.Request.Method == http.MethodConnect {
```

Comment:
I'm not sure this should be here.

Failed connect is free to return response as any other method.

Reply:
OK, I will drop that.